$OpenBSD$

* We don't use -ldl
* Don't use platform specific paths
* Honour CFLAGS and CC

--- lp_solve/ccc.orig	Sun Jan 25 18:39:03 2009
+++ lp_solve/ccc	Wed Apr 18 16:50:34 2012
@@ -1,6 +1,6 @@
 :
 src='../lp_MDO.c ../shared/commonlib.c ../colamd/colamd.c ../shared/mmio.c ../shared/myblas.c ../ini.c ../fortify.c ../lp_rlp.c ../lp_crash.c ../bfp/bfp_LUSOL/lp_LUSOL.c ../bfp/bfp_LUSOL/LUSOL/lusol.c ../lp_Hash.c ../lp_lib.c ../lp_wlp.c ../lp_matrix.c ../lp_mipbb.c ../lp_MPS.c ../lp_params.c ../lp_presolve.c ../lp_price.c ../lp_pricePSE.c ../lp_report.c ../lp_scale.c ../lp_simplex.c lp_solve.c ../lp_SOS.c ../lp_utils.c ../yacc_read.c'
-c=cc
+c=${CC}
 
 #determine platform (32/64 bit)
 >/tmp/platform.c
@@ -11,7 +11,7 @@ $c /tmp/platform.c -o /tmp/platform
 PLATFORM=`/tmp/platform`
 rm /tmp/platform /tmp/platform.c >/dev/null 2>&1
 
-mkdir bin bin/$PLATFORM >/dev/null 2>&1
+mkdir bin >/dev/null 2>&1
 
 math=-lm
 
@@ -28,14 +28,14 @@ else NOISNAN=-DNOISNAN
 fi
 rm /tmp/isnan.c /tmp/isnan >/dev/null 2>&1
 
-opts='-O3'
+opts=${CFLAGS}
 
 def=
 if [ "$PLATFORM" = "SCO_UNIX" ]
 then opts='-O0'
      def='-dy -K PIC -DLLONG=long'
      dl=-ldl
-else dl=-ldl
+else dl=-lcompat
 fi
 
-$c -I.. -I../bfp -I../bfp/bfp_LUSOL -I../bfp/bfp_LUSOL/LUSOL -I../colamd -I../shared $opts $def $NOISNAN -DYY_NEVER_INTERACTIVE -DPARSER_LP -DINVERSE_ACTIVE=INVERSE_LUSOL -DRoleIsExternalInvEngine $src -o bin/$PLATFORM/lp_solve $math $dl
+$c -I.. -I../bfp -I../bfp/bfp_LUSOL -I../bfp/bfp_LUSOL/LUSOL -I../colamd -I../shared $opts $def $NOISNAN -DYY_NEVER_INTERACTIVE -DPARSER_LP -DINVERSE_ACTIVE=INVERSE_LUSOL -DRoleIsExternalInvEngine $src -o bin/lp_solve $math $dl
